File: algorithm/6. ZigZag Conversion/6.cpp
class Solution {
public:
    string convert(string s, int numRows) {
        if (numRows == 1) return s;
        string sss[numRows];
        int gap = numRows-2, i = 0; 
        while( i < s.length() )
        {
            for (int j = 0; j < numRows && i < s.size() ; j ++)
                sss[j]+=s[i++];
            for (int j = gap; j > 0 && i < s.size() ; j --)
                sss[j]+=s[i++];
        }
        string result="";
        for (int j =0; j < numRows; j++)
            result+= sss[j];
        return result;
    }
};

File: algorithm/6. ZigZag Conversion/6.py
class Solution(object):
    def convert(self, s, numRows):
        """
        :type s: str
        :type numRows: int
        :rtype: str
        """
        if numRows == 1:
            return s
        sss = [""]*numRows
        i=0
        gap = numRows - 2
        while i < len(s):
            for j in xrange(numRows):
                if i >= len(s):
                    break
                sss[j]+=s[i]
                i+=1
            for j in range(gap, 0,-1  ):
                if i >= len(s):
                    break
                sss[j]+=s[i]
                i+=1
        s=""
        for j in xrange(numRows):
            s+=sss[j]
        return s
